File: src/main/java/org/telran/lecture_4_d_and_c/hw/TwoArrays.java
package org.telran.lecture_4_d_and_c.hw;

import java.util.PriorityQueue;

public class TwoArrays {
    public static void main(String[] args) {

    }
    public static int getElementTwoSortedArrays(int[] arr1, int[] arr2, int index) {
        PriorityQueue<Integer> priorityQueue = new PriorityQueue<>();
        // Pushing elements for array arr1 to min-heap
        for (int element : arr1) {
            priorityQueue.offer(element);
        }
        // Pushing elements for array arr2 to min-heap
        for (int element : arr2) {
            priorityQueue.offer(element);
        }
        // Popping-out K-1 elements
        while (index-- > 1) {
            priorityQueue.remove();
        }
        return priorityQueue.peek();
    }
}

File: src/main/java/org/telran/lecture_4_d_and_c/hw/TwoArrays.js
function getElementTwoSortedArrays(arr1, arr2, key) {
    let length1 = arr1.length;
    let length2 = arr2.length;
    let sorted1 = Array(length1 + length2).fill(0);
    let i = 0, j = 0, d = 0;
    while (i < length1 && j < length2) {
        if (arr1[i] < arr2[j])
            sorted1[d++] = arr1[i++];
        else
            sorted1[d++] = arr2[j++];
    }
    while (i < length1)
        sorted1[d++] = arr1[i++];
    while (j < length2)
        sorted1[d++] = arr2[j++];
    return sorted1[key - 1];
}

let arr1 = [ 2, 3, 6, 7, 9 ];
let arr2 = [ 1, 4, 8, 10 ];
let key = 5;
document.write(getElementTwoSortedArrays(arr1, arr2, key));
 